src/stream/ngx_stream_ssl_module.c

```diff
@@ -636,6 +653,10 @@ ngx_stream_ssl_merge_conf(ngx_conf_t *cf, void *parent, void *child)
 
     ngx_conf_merge_str_value(conf->ciphers, prev->ciphers, NGX_DEFAULT_CIPHERS);
 
+#if (NGX_STREAM_SNI)
+    ngx_conf_merge_value(conf->sni_force, prev->sni_force, 0);
+    if (!conf->listen)
```

Comment:
wrong added line ` if (!conf->listen)`, it make conf->ssl.log NULL.

Then the following nginx.conf will trigger a segfault when nginx starts.

```
stream {
    server {
        listen 127.0.0.1:8980 reuseport ssl udp;
        ssl_protocols dtlsv1;
        ssl_certificate_key localhost.key;
        ssl_certificate localhost.crt;
        return "ok";
    }
}
```

Reply:
fixed in https://github.com/alibaba/tengine/pull/1684